### PR TITLE
fix(canvas): stop Save to Canvas duplicating byte-identical sketches (cave-spq7)

### PR DESCRIPTION
**Pull request by BunsDev**
## What

**Bug** (cave-spq7, found by driving the running app): every *Save to Canvas* mints a fresh `art-${uuid}`, and `upsertCanvasArtifact` deduped **by id only** — so re-saving an unchanged sketch (re-save from a gallery-opened viewer, a double save from chat, an add-tile re-keep) appended a byte-identical twin tile. The live store on this machine had two such dup pairs (identical `code`, ids minutes apart), visible as twin cards in the Canvas tab.

## Fix

`upsertCanvasArtifact` content-dedupes at the single choke point (all saves flow through `POST /api/canvas` → here):

- after the id filter, a record with the **same `kind` + `code`** is updated in place — the incumbent keeps its **id** (and with it `createdAt` and its saved canvas position); the caller's `title`/`prompt`/`updatedAt` are taken
- same-**id** replaces (the refine path) still win over content dedupe
- same code under a **different kind** stays a separate artifact
- the function returns the id the save **settled under**; the route reports it as `savedId`, and `CanvasAddTile` highlights that tile instead of the client-minted id a dedupe discarded

## Verification

- Behavioral tests against a `COVEN_CAVE_HOME` temp store (dedupe, id/createdAt survival, kind separation, same-id replace, junk payload, delete) + wiring pins for `savedId`. **Test-hygiene note:** the env var is now set *before* a dynamic import of `cave-canvas.ts` — a hoisted static import evaluated `CANVAS_PATH` against the real store, and an earlier draft's first run leaked one record into live data (removed immediately, backup kept).
- `pnpm test:app` — 765 test files pass; `pnpm typecheck` — clean
- Canvas surface verified in the running app (v0.1.3 build): gallery, viewer modal (Canvas/Code tabs), add-tile composer all healthy; zero console/page errors

## Refs

- Bead: cave-spq7

### CODE REVIEW

File: src/components/canvas-add-tile.tsx [hunk 1]
```diff
@@ -189,11 +189,13 @@ export function CanvasAddTile({ familiarId, hero = false, onSaved }: {
         body: JSON.stringify({ artifact }),
       });
       if (!res.ok) throw new Error(String(res.status));
-      const data = (await res.json()) as { artifacts?: CanvasArtifact[] };
+      const data = (await res.json()) as { artifacts?: CanvasArtifact[]; savedId?: string };
```
[the remark this answers is not in the document]
Comment (CompleteDotTech) [reply src/components/canvas-add-tile.tsx:192]: Addressed in follow-up PR #3400 because #3392 merged while this review pass was still in progress. The response type now includes `null`; the follow-up also fixes and tests the same-id/twin collision so explicit updates remain authoritative.

**Comment by CompleteDotTech** [comment]
Follow-up review found and fixed two post-merge issues in #3400:

- the unresolved nullable `savedId` review request is addressed
- same-id replacement now remains authoritative even when edited content matches another stored artifact; previously that edge case settled under the twin id and deleted the explicitly updated record

Local exact-tree verification on #3400: focused Canvas persistence test, typecheck, tests-wired (1,039), full app suite (765), and diff check all pass. I’ll keep the follow-up draft until its GitHub checks and review are clean.

**Comment by CompleteDotTech** [comment]
Completed the post-merge follow-up in #3400 (squash `0586b059`): the nullable `savedId` contract and the same-id/twin collision are now fixed on `main` with regression coverage. Exact-head local and GitHub CI evidence is recorded on #3400; all required platform, sidecar, CodeQL, and CodeQL Advanced checks passed. I will resolve the original review thread as soon as GitHub’s GraphQL rate-limit window resets.
